org.eclipse.m2e.pde.target.tests/src/org/eclipse/m2e/pde/target/tests/MavenFeatureTest.java

```diff
@@ -0,0 +1,105 @@
+ *   Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.target.tests;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.pde.core.target.ITargetDefinition;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class MavenFeatureTest extends AbstractMavenTargetTest {
+	@Parameter(0)
+	public Boolean includeSource;
+
+	@Parameters(name = "includeSource={0}")
+	public static List<Boolean> dependencyConfigurations() {
+		return List.of(false, true);
+	}
+
+	@Test
+	public void testLocationFeatureGeneration() throws CoreException {
+		ITargetDefinition target = resolveMavenTarget(String.format(
+				"""
+						<location includeDependencyDepth="infinite" includeDependencyScopes="compile" includeSource="%s" missingManifest="generate" type="Maven">
+							<feature id="my.junit.feature" label="My Junit Feature" provider-name="Me Inc." version="1.2.3.qualifier">
+								<description url="http://www.example.com/description">
+									This is a great feature
+								</description>
+								<copyright url="http://www.example.com/copyright">
+									My copyright description.
+								</copyright>
+								<license url="http://www.example.com/license">
+									No license granted.
+								</license>
+								<plugin download-size="0" id="slf4j.api" install-size="0" unpack="false" version="0.0.0"/>
+							</feature>
+							<dependencies>
+								<dependency>
+									<groupId>org.junit.jupiter</groupId>
+									<artifactId>junit-jupiter-api</artifactId>
+									<version>5.9.3</version>
+									<type>jar</type>
+								</dependency>
+							</dependencies>
+						</location>
+						""",
+				includeSource));
+		assertTrue(target.getStatus().isOK());
+		List<ExpectedBundle> expectedBundles = List.of( //
+				junitJupiterAPI(), //
+				junitPlatformCommons(), //
+				apiGuardian(), opentest4j());
+		assertTargetBundles(target, includeSource ? withSourceBundles(expectedBundles) : expectedBundles);
+		List<ExpectedFeature> expectedFeature = List.of(generatedFeature("my.junit.feature", "1.2.3.qualifier", List.of(//
+				featurePlugin("junit-jupiter-api", "5.9.3"), //
+				featurePlugin("junit-platform-commons", "1.9.3"), //
+				featurePlugin("org.apiguardian.api", "1.1.2"), //
+				featurePlugin("org.opentest4j", "1.2.0"), //
+				featurePlugin("slf4j.api", null))));
+		assertTargetFeatures(target, includeSource ? withSourceFeatures(expectedFeature) : expectedFeature);
+	}
+
+	@Test
+	public void testPomFeatureGeneration() {
+		// TODO Auto-generated method stub
+	}
+
+	@Test
+	public void testFeatureArtifact() {
+		// TODO Auto-generated method stub
```

Comment:
@laeubi do you now a Eclipse Feature available as artifact from Maven-Central?

Reply:
You can search for "feature" in central and look for items ending with ".feature" and having a jar, here is an example of an very old one:

https://central.sonatype.com/artifact/org.drools/org.drools.eclipse.task.feature/5.4.0.CR1/versions